tests/requestparsing/contentlengthandtransferencoding_test.go

```diff
@@ -0,0 +1,347 @@
+package requestparsing
+
+import (
+	"bytes"
+	"context"
+	"io/ioutil"
+	"net/http"
+	"testing"
+
+	"github.com/google/go-safeweb/testing/requesttesting"
+
+	"github.com/google/go-cmp/cmp"
+)
+
+func TestContentLengthTransferEncoding(t *testing.T) {
+	type testWant struct {
+		headers          map[string][]string
+		contentLength    int64
+		transferEncoding []string
+		body             string
+	}
+
+	var tests = []struct {
+		name    string
+		request []byte
+		want    testWant
+	}{
+		{
+			name: "ContentLength",
+			request: []byte("GET / HTTP/1.1\r\n" +
+				"Host: localhost:8080\r\n" +
+				"Content-Length: 5\r\n" +
+				"\r\n" +
+				"ABCDE\r\n" +
+				"\r\n"),
+			want: testWant{
+				headers:          map[string][]string{"Content-Length": []string{"5"}},
+				contentLength:    5,
+				transferEncoding: nil,
+				body:             "ABCDE",
+			},
+		},
+		{
+			name: "ContentButNoContentLength",
```

Comment:
General comment: these names do not have to be camel case. I usually see regular strings with spaces.